castle/cms/cdn.py

```diff
@@ -0,0 +1,31 @@
+from plone import api
+
+
+class cdn(object):
+
+    def __init__(self, hostname, port=80, path=''):
+        self.hostname = api.portal.get_registry_record('castle.cdn_alternate_domain') or hostname
```

Comment:
hostname, port, and path here -- as configuration values, should be condensed to just one value -- a URL prefix.

The prefix should wholly replace anything in the url before the path.

IE I should be able to enter, in one field `https://cdn.example.site/prefix`

and the normal url, say `https://example.site/an/example/path`, would be transformed to `https://cdn.example.site/prefix/an/example/path`